src/databricks/labs/ucx/source_code/path_lookup.py

```diff
@@ -27,19 +33,59 @@ def change_directory(self, new_working_directory: Path) -> PathLookup:
         return PathLookup(new_working_directory, self._sys_paths)
 
     def resolve(self, path: Path) -> Path | None:
-        if path.is_absolute() and path.exists():
-            # eliminate “..” components
-            return path.resolve()
-        for parent in self.library_roots:
-            absolute_path = parent / path
+        try:
+            if path.is_absolute() and path.exists():
+                # eliminate “..” components
+                return path.resolve()
+        except PermissionError:
+            logger.warning(f"Permission denied to access {path}")
+            return None
+        for library_root in self.library_roots:
+            resolved_path = self._resolve_library_root(library_root, path)
+            if resolved_path is not None:
+                return resolved_path
+        return None
```

Comment:
```suggestion
                return path.resolve()
                for library_root in self.library_roots:
                    resolved_path = self._resolve_library_root(library_root, path)
                    if resolved_path is not None:
                        return resolved_path
                    return None
        except PermissionError as e:
            logger.warning(f"Permission denied to access {path}: {e}")
            return None
```
have just one permissions error catch in this public method. otherwise the code is a bit unreadable.